docs(tab): update tab docs and a11y (#5586)
adobe/spectrum-web-components: packages/tabs/tab.md

## Overview

> An `<sp-tab>` element surfaces a `label` attribute to serve as its default text content when none is available in the DOM. An icon may be assigned to the element via the `icon` slot; e.g. `<sp-tab><svg slot="icon" label="Tab with icon">...</svg></sp-tab>`. Associate an `<sp-tab>` element with the `<sp-tab-panel>` that represents its content with the `value` attribute.

Reviewer: The default slot and the `label` attribute both can set the label internally with a `<label>` element.

[![How big is this package in your project?](https://img.shields.io/bundlephobia/minzip/@spectrum-web-components/tabs?style=for-the-badge)](https://bundlephobia.com/result?p=@spectrum-web-components/tabs)

```bash
yarn add @spectrum-web-components/tabs
```

Import the side effectful registration of `<sp-tab>` via:

```js
import '@spectrum-web-components/tabs/sp-tab.js';
```

When looking to leverage the `Tab` base class as a type and/or for extension purposes, do so via:

```js
import { Tab } from '@spectrum-web-components/tabs';
```

### Anatomy

The `<sp-tab>` element consists of the following:

- a label, so that users know which tab panels content will be shown
- an optional icon
- a value that programmatically associates the tab with its corresponding tab panel

```html
<sp-tabs>
    <sp-tab label="Tab cola" value="tab-cola">
        <sp-icon-checkmark slot="icon"></sp-icon-checkmark>
    </sp-tab>
    <sp-tab-panel value="tab-cola">
        Tab (or TaB) was a diet cola soft drink from The Coca-Cola Company, introduced in 1963. It is no longer produced.
    </sp-tab-panel>
</sp-tab>
```

### Options

#### Icon only

> For tabs that have an icon and no visible label, the icon should have a `label` attribute in order to set the `aria-label` of the icon. Icons should not be used just as decoration. If the tab item does not have a visible label, it must still have a tooltip to disclose the label.

Reviewer: We could provide an example here that uses tooltip and a `label` attribute but we should also make very clear that a tooltip is no substitute for a label.

```html
<sp-tabs>

>     <sp-tab value="complete">
>         <sp-icon-checkmark
>             slot="icon"
>             label="Checking your work"
>         ></sp-icon-checkmark>
>     </sp-tab>

Reviewer: ```suggestion
    <sp-tab value="comments">
        <sp-tooltip slot="icon">
            <sp-icon-comment
                slot="icon"
                label="Viewing your comments"
             ></sp-icon-checkmark>
             Comments
        </sp-tooltip>
        <sp-icon-checkmark
            slot="icon"
            label="Checking your work"
        ></sp-icon-checkmark>
        <sp-tab-panel value="comments">
            A screenreader will read this tab as "Viewing your comments"
        </sp-tab-panel>
    </sp-tab>
```

[![See it on NPM!](https://img.shields.io/npm/v/@spectrum-web-components/tabs?style=for-the-badge)](https://www.npmjs.com/package/@spectrum-web-components/tabs)
[![How big is this package in your project?](https://img.shields.io/bundlephobia/minzip/@spectrum-web-components/tabs?style=for-the-badge)](https://bundlephobia.com/result?p=@spectrum-web-components/tabs)

```zsh
yarn add @spectrum-web-components/tabs
```

Import the side effectful registration of `<sp-tab>` via:

```ts
import '@spectrum-web-components/tabs/sp-tab.js';
```

When looking to leverage the `Tab` base class as a type and/or for extension purposes, do so via:

```ts
import { Tab } from '@spectrum-web-components/tabs';
```

### Anatomy

The `<sp-tab>` element consists of a label and an optional icon.

```html
<sp-tab label="Label">
    <sp-icon-checkmark slot="icon"></sp-icon-checkmark>
</sp-tab>
```

### Options

#### Without a label

> ```html

Reviewer: ````suggestion
For tabs that have an icon and no visible label, the icon should have a `label`.
```html
````

> ```html

Reviewer: ````suggestion
For tabs that have an icon and no visible label, the icon should have a `label`.
```html
````

<sp-tab value="complete">

>     <sp-icon-checkmark slot="icon"></sp-icon-checkmark>

Reviewer: ```suggestion
    <sp-icon-checkmark slot="icon" label="Checking your work"></sp-icon-checkmark>
```

[![How big is this package in your project?](https://img.shields.io/bundlephobia/minzip/@spectrum-web-components/tabs?style=for-the-badge)](https://bundlephobia.com/result?p=@spectrum-web-components/tabs)

```bash
yarn add @spectrum-web-components/tabs
```

Import the side effectful registration of `<sp-tab>` via:

```ts
import '@spectrum-web-components/tabs/sp-tab.js';
```

When looking to leverage the `Tab` base class as a type and/or for extension purposes, do so via:

```ts
import { Tab } from '@spectrum-web-components/tabs';
```

### Anatomy

The `<sp-tab>` element consists of a label and an optional icon.

```html
<sp-tab label="Label">
    <sp-icon-checkmark slot="icon"></sp-icon-checkmark>
</sp-tab>
```

### Options

#### Icon only

For tabs that have an icon and no visible label, the icon should have an `aria-label`. Icons should not be used just as decoration. If the tab item does not have a visible label, it must still have a tooltip to disclose the label.

```html
<sp-tab value="complete" aria-label="Tab with checkmark">
    <sp-icon-checkmark slot="icon"></sp-icon-checkmark>
</sp-tab>
```

#### Label only

```html
<sp-tab value="label" label="Label"></sp-tab>

> ```

Reviewer 2: Do you mean both the `value` and `label` attributes?

I added a line about what the difference was, if that helps.

And then I also updated some of the Accessibility best practices below to be a bit more explicit, it looks like you can use the `label` attribute or put the text in the slot content.

> ```

Reviewer: See my suggestion above

[![See it on NPM!](https://img.shields.io/npm/v/@spectrum-web-components/tabs?style=for-the-badge)](https://www.npmjs.com/package/@spectrum-web-components/tabs)
[![How big is this package in your project?](https://img.shields.io/bundlephobia/minzip/@spectrum-web-components/tabs?style=for-the-badge)](https://bundlephobia.com/result?p=@spectrum-web-components/tabs)

```bash
yarn add @spectrum-web-components/tabs
```

Import the side effectful registration of `<sp-tab>` via:

```js
import '@spectrum-web-components/tabs/sp-tab.js';
```

When looking to leverage the `Tab` base class as a type and/or for extension purposes, do so via:

```js
import { Tab } from '@spectrum-web-components/tabs';
```

### Anatomy

The `<sp-tab>` element consists of a label and an optional icon.

```html
<sp-tab label="Label">
    <sp-icon-checkmark slot="icon"></sp-icon-checkmark>
</sp-tab>
```

### Options

#### Icon only

For tabs that have an icon and no visible label, the icon should have an `aria-label`. Icons should not be used just as decoration. If the tab item does not have a visible label, it must still have a tooltip to disclose the label.

```html

> <sp-tab value="complete" aria-label="Tab with checkmark">
>     <sp-icon-checkmark
>         slot="icon"
>         label="Checking your work"
>     ></sp-icon-checkmark>
> </sp-tab>

Reviewer: For tabs with icon only, you can label the tab using the icon's `label` attribute:

```suggestion
    <sp-tabs>
        <sp-tab value="complete">
            <sp-icon-checkmark
                slot="icon"
                label="Checking your work"
            ></sp-icon-checkmark>
        </sp-tab>
       <sp-tab-panel value="complete">
           A screenreader will read this tab as "Checking your work"
       </sp-tab-panel>
    <sp-tabs>
```

```

#### Label only

```html
<sp-tab value="label" label="Label"></sp-tab>
```

The `label` attribute provides the visible text content and accessible name for screen readers. The `value` attribute links the tab to its corresponding `sp-tab-panel` via matching values.

### States

In order to activate the `<sp-tab>` element's interactive states, it must be used within an `<sp-tabs>` element.

#### Selected state

The tab is currently active and its associated panel is visible.

```html

> <sp-tabs>
>     <sp-tab selected="label" label="Label" value="1"></sp-tab>
> </sp-tabs>

Reviewer: ```suggestion
<sp-tabs>
    <sp-tab selected="label" label="Label" value="1"></sp-tab>
    <sp-tab-panel value="1">This tab is selected.</sp-tab-panel>
</sp-tabs>
```

```

#### Focused state

The tab has keyboard focus. All tabs can receive focus through keyboard navigation, except when the tab is `disabled`.

```html
<sp-tabs selected="2">
    <sp-tab label="Tab 1" value="1"></sp-tab>
    <sp-tab label="Tab 2" value="2"></sp-tab>
    <sp-tab label="Tab 3" value="3"></sp-tab>
</sp-tabs>

> ```

Reviewer: Since we can't show focused without stealing keyboard focus, we could probably remove the example here.

### Options

#### Icon only

For tabs that have an icon and no visible label, the icon should have a `label`. Icons should not be used just as decoration. If the tab item does not have a visible label, it must still have a tooltip to disclose the label.

```html
<sp-tab value="complete">
    <sp-icon-checkmark
        slot="icon"
        label="Checking your work"

>         aria-label="Tab w/ checkmark"

Reviewer 3: I'd have to check with Nikki again, but would we want the `label` and `aria-label` to be the same, that way screen reader users get the same experience & information? 

When I tested this in Safari, VoiceOver also announces: "Tab w slash checkmark" instead of what this was probably going for "Tab with checkmark."

```html
<sp-tab value="complete" aria-label="Tab with checkmark">
    <sp-icon-checkmark
        slot="icon"
        label="Checking your work"

>     ></sp-icon-checkmark>

Reviewer 2: @nikkimk would love some feedback on this one after Marissa's [feedback](https://github.com/adobe/spectrum-web-components/pull/5586#discussion_r2208270711)! Do we want it like this, or how it was previously, or something else?

#### Icon only

> For tabs that have an icon and no visible label, the icon should have a `label` attribute in order to set the `aria-label` of the icon. Icons should not be used just as decoration. If the tab item does not have a visible label, it must still have a tooltip to disclose the label.

Reviewer 2: Correct me if I'm wrong, my understanding of how icon works from checking the code and the documentation is that if the `label` attribute is set, it sets `aria-label` on `<sp-icon>` for us? I made a few small updates to the documentation to reflect this, but if that's wrong, please let me know.

    <sp-tab-panel value="4">Content for Tab 4 is selectable</sp-tab-panel>
</sp-tabs>
```

### Behaviors

#### Selection

- Clicking a tab selects it and shows its associated panel
- Only one tab can be selected at a time within a tab group
- Disabled tabs cannot be selected

#### Keyboard Navigation

- `Tab`: Move focus to the next focusable element
- `Arrow Keys`: Navigate between tabs in the group
- `Enter` or `Space`: Select the focused tab

> 

Reviewer: I'd move the keyboard navigation section to accessibility to be consistent with other components that include this information.

#### Events

- `click`: Fired when the tab is clicked
- `keydown`: Fired when a key is pressed while the tab has focus

### Accessibility

#### Best Practices

> - Always provide meaningful text content via `label` attribute or text content

Reviewer: ```suggestion
- Always provide meaningful text content via `label` attribute or text content. In rare cases where an icon provides enough content, use the icon's `label` attribute.
```

- Use descriptive `value` attributes that clearly identify the tab's purpose
- Ensure tab labels are concise but descriptive

> - When using icons, provide appropriate `aria-label` attributes

Reviewer: ```suggestion
```

> - When using icons, provide appropriate `aria-label` attributes

Author: I'll add it
